components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-hardware-account-modal.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory, useParams } from 'react-router'
+
+// utils
+import { getLocale } from '$web-common/locale'
+
+// options
+import { CreateAccountOptions } from '../../../../options/create-account-options'
+
+// types
+import { CreateAccountOptionsType, WalletRoutes, WalletState } from '../../../../constants/types'
+
+// actions
+import { WalletPageActions } from '../../../../page/actions'
+
+// components
+import { DividerLine } from '../../../extension'
+import PopupModal from '..'
+import { HardwareWalletConnect } from './hardware-wallet-connect'
+import { SelectAccountType } from './select-account-type/select-account-type'
+
+// style
+import {
+  StyledWrapper
+} from './style'
+
+interface Params {
+  accountTypeName: string
+}
+
+export const AddHardwareAccountModal = () => {
+  // routing
+  const history = useHistory()
+  const { accountTypeName } = useParams<Params>()
+
+  // redux
+  const dispatch = useDispatch()
+  const isFilecoinEnabled = useSelector(({ wallet }: { wallet: WalletState }) => wallet.isFilecoinEnabled)
+  const isSolanaEnabled = useSelector(({ wallet }: { wallet: WalletState }) => wallet.isSolanaEnabled)
```

Comment:
Can these be combined and deconstructed?

Reply:
selectors should only return one value to reduce chances or re-rendering